packages/cli/src/databases/repositories/workflow.repository.ts

```diff
@@ -13,16 +13,38 @@ import type {
 import type { ListQuery } from '@/requests';
 import { isStringArray } from '@/utils';
 
+import { FolderRepository } from './folder.repository';
+import type { Folder } from '../entities/folder';
 import { TagEntity } from '../entities/tag-entity';
 import { WebhookEntity } from '../entities/webhook-entity';
 import { WorkflowEntity } from '../entities/workflow-entity';
 import { WorkflowTagMapping } from '../entities/workflow-tag-mapping';
 
+type ResourceType = 'folder' | 'workflow';
+
+type WorkflowAndFolderUnion = {
```

Comment:
Maybe this could be named to be explicit that it's a row returned by a query

```suggestion
type WorkflowFolderUnionRow = {
```